apis/r/vignettes/soma-experiment-queries.Rmd

```diff
@@ -66,17 +66,85 @@ Number of variables:
 query$n_vars
 ```
 
-We can also extract any data component from the experiment. Here we'll read in the `obs` data frame from the query.
+We can also extract any data component from the experiment. Here we'll read in the `obs` 
+data frame from the query using `obs()` which returns an iterator of `arrow::Table`. The iterator is useful 
+when the data is too large to load in memory allowing to stream the data in chunks. This 
+applies to `var()` as well.
+
+To load the data in memory you can concatenate all chunks of the iterator as shown below.
 
 ```{r}
-query$obs()
+iterator <- query$obs()
+obs <- iterator$concat()
+obs
+```
+As a reminder `arrow:Table` can be easily cast into a `tibble`
+
+```{r}
+as.data.frame(obs)
+```
+
+Alternatively, you can use the iterator, which retrieves data in chunks that are smaller than 
+the `soma.init_buffer_bytes` context field. You can use the iterator's method `$read_next()` to 
+load a chunk in memory. In this example the data is so small that it all fits in one chunk.
+
+```{r}
+iterator <- query$obs()
+iterator$read_next()
+```
+
+In this example the full `obs` table is relatively small and fits all in one chunk.
+
+For a bigger `SOMADataFrame` you can check if the iteration has finished by checking the logical `$read_complete()`.
+
+Here we demonstrate by creating a new iterator.
+
+```{r}
+iterator <- experiment$obs$read()
+iterator$read_complete()
+```
+```{r}
+iterator$read_next()
```

Comment:
```suggestion
```{r}
```

Suggested simplification: I don't think we gain anything by creating a new iterator rather than just calling `read_complete()` on the existing one from above.

Reply:
The only reason I did it is because the previous call returns `True` for `read_complete()` since one iteration is enough. I'm happy to remove if you think this not very valuable.

Reply:
Ah okay, thanks for the clarification! I'm on the fence about whether it's worth including or not but I don't feel as strongly about it with the extra info you provided, so your call!